src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -616,10 +616,19 @@ private List<LaunchCommand> GetInitializeCommands()
 
                     CheckCygwin(commands, localLaunchOptions);
 
-                    // ClrDbg doesn't need it.
-                    if (this.MICommandFactory.Mode != MIMode.Clrdbg)
+                    // ClrDbg doesn't need -file-exec-and-symbols set.
+                    if (this.MICommandFactory.Mode == MIMode.Gdb)
                     {
-                        this.AddExecutablePathCommand(commands);
+                        if (_launchOptions is UnixShellPortLaunchOptions)
```

Comment:
I wanted it to be specific to *nix and GDB. I'll write a comment for now.

---
In reply to: [75935168](https://github.com/Microsoft/MIEngine/pull/422#discussion_r75935168) [](ancestors = 75935168)